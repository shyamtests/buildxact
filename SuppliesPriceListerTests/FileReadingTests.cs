using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using buildxact_supplies;
using buildxact_supplies.DataSources;
using Microsoft.Extensions.Configuration;
using Newtonsoft.Json;
using Xunit;

namespace SuppliesPriceListerTests
{
    [UseReporter(typeof(DiffReporter))]
    public class FileReadingTests
    {
        [Fact]
        public void HumphriesReadsEmbeddedFilesAndConvertsToBuildingSupplies()
        {
            var supplies = new HumphriesSupplies().GetBuildingSupplies();
            
            Approvals.VerifyJson(JsonConvert.SerializeObject(supplies));
        }
        
        [Fact]
        public void MegaCorpReadsEmbeddedFilesAndConvertsToBuildingSupplies()
        {

            var myConfiguration = new Dictionary<string, string>
            {
                {"audUsdExchangeRate", "1"}
            };

            
            var configuration = new ConfigurationBuilder()
                .AddInMemoryCollection(myConfiguration)
                .Build();
            
            
            var supplies = new MegacorpSupplies(new CurrencyConverter(configuration)).GetBuildingSupplies();
            
            Approvals.VerifyJson(JsonConvert.SerializeObject(supplies));
        }
        
        // Prices for this approval test should be double the previous
        [Fact]
        public void MegaCorpReadsEmbeddedFilesAndConvertsToBuildingSuppliesAndConvertsToAud()
        {

            var myConfiguration = new Dictionary<string, string>
            {
                {"audUsdExchangeRate", "0.5"}
            };

            
            var configuration = new ConfigurationBuilder()
                .AddInMemoryCollection(myConfiguration)
                .Build();
            
            
            var supplies = new MegacorpSupplies(new CurrencyConverter(configuration)).GetBuildingSupplies();
            
            Approvals.VerifyJson(JsonConvert.SerializeObject(supplies));
        }
    }
}
