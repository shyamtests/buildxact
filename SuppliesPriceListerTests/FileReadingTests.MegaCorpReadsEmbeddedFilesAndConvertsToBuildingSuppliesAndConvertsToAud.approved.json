[
    {
        "Id": "1",
        "Name": "100 x 200 x 20mpa Internal Beam",
        "Price": 80.0
    },
    {
        "Id": "0",
        "Name": "100 x 350 Thickened Edge",
        "Price": 70.0
    },
    {
        "Id": "2",
        "Name": "1000 x 1000 x 350 x 20mpa Square Pad Footing",
        "Price": 24.5
    },
    {
        "Id": "3",
        "Name": "100mm x 25mpa Slab",
        "Price": 404.7
    },
    {
        "Id": "4",
        "Name": "100mm x 32 mpa Slab",
        "Price": 45.0
    },
    {
        "Id": "6",
        "Name": "100 x 200 x 20mpa Internal Beam",
        "Price": 4.0
    },
    {
        "Id": "7",
        "Name": "11 x 200 Trench Mesh",
        "Price": 29.92
    }
]